### PR TITLE
Add note to manhole.md about bind_address when using with docker

### PR DESCRIPTION
**Pull request by maquis196**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by richvdh** [comment]
@maquis196 it's helpful if you can link issue numbers in the PR description.

Fixes #8508

**Comment by maquis196** [comment]
Okay, yeah its in the commit but that makes sense!

File: docs/manhole.md [hunk 1]
```diff
@@ -18,6 +18,9 @@ listeners:
 (`bind_addresses` in the above is important: it ensures that access to the
 manhole is only possible for local users).
 
+**Docker Note**
+If you are using docker, you will need to use 0.0.0.0 for the bind_addresses above. This is so docker can forward traffic to synapse.
```
Comment (richvdh) [inline docs/manhole.md:22]: So `bind_addreses: 0.0.0.0` ? Please could you give an explicit example?

**Review by richvdh — CHANGES_REQUESTED** [review]

**Comment by richvdh** [comment]
@maquis196 see also https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords. Github has magic for tracking this.

**Comment by ShadowJonathan** [comment]
Thanks for PRing a fix for my issue! :+1:

File: docs/manhole.md [hunk 2]
```diff
@@ -18,6 +18,16 @@ listeners:
 (`bind_addresses` in the above is important: it ensures that access to the
 manhole is only possible for local users).
 
+**Docker Note**
+If you are using docker, please ensure you use the following example;
```
Comment (richvdh) [inline docs/manhole.md:22]:
```suggestion
If you are using Docker, set `bind_addresses` to `['0.0.0.0']` as shown
in the following example, to allow connections from the docker host:
```

File: docs/manhole.md [hunk 3]
```diff
@@ -18,6 +18,16 @@ listeners:
...
+
+```yaml
+listeners:
+  - port: 9000
+    bind_addresses: ['0.0.0.0']
+    type: manhole
+```
+
```
Comment (richvdh) [inline docs/manhole.md:30]: can you say something about needing to forward the port with `docker run` as well?

File: docs/manhole.md [hunk 4]
```diff
@@ -18,6 +18,16 @@ listeners:
...
 Note that this will give administrative access to synapse to **all users** with
```
Comment (richvdh) [inline docs/manhole.md:31]: now it looks a bit like this only applies to docker, which isn't the case. I think you need more headings.

**Review by richvdh — CHANGES_REQUESTED** [review]

**Comment by maquis196** [comment]
HAve done some reordering and put an example in to ensure docker is localhost as well on purpose.

File: docs/manhole.md (hunk 3 above)
Comment (maquis196) [reply docs/manhole.md:54]: I've added a simple "example" config that anyone who wants to use manhole and is using docker should clearly understand.

File: docs/manhole.md [hunk 5]
```diff
@@ -5,22 +5,54 @@ The "manhole" allows server administrators to access a Python shell on a running
...
 listeners:
   - port: 9000
-    bind_addresses: ['::1', '127.0.0.1']
+    bind_addresses: ['0.0.0.0']
     type: manhole
 ```
 
-(`bind_addresses` in the above is important: it ensures that access to the
-manhole is only possible for local users).
+You will then need to change the docker command to the following to include the
```
Comment (ShadowJonathan) [inline docs/manhole.md:30]:
```suggestion
When using `docker run` to start the server, you will then need to change the command to the following to include the
```

**Review by richvdh — APPROVED** [review]
this is nice and clear now. Thanks!

**Pull request merged**